pandas/io/formats/style.py

```diff
@@ -930,6 +930,25 @@ def hide_columns(self, subset):
     # A collection of "builtin" styles
     # -----------------------------------------------------------------------
 
+    def format_null(self, na_rep="-"):
+        """
+        Format the text displayed for missing values.
+
+        .. versionadded:: 1.0.0
+
+        Parameters
+        ----------
+        na_rep : str
+
+        Returns
+        -------
+        self : Styler
+        """
+        self.format(
```

Comment:
This looks like it will overwrite the formatting of a previously applied formatter for non-NA values. Something like

```python
df.style.format("hi-{}".format).format_null()
```

is that the case?

Reply:
Thanks @TomAugspurger, I like the name `format_na`! and yes, I was intended to make an overwriting implementation. Actually, I have considered the `set_*` approach as you, but it seems confusing for the case:
```
df.style.format('{:.2%}').set_na_format('-') # got 'nan%' instead of '-'
```
I've got a new idea, how about interface like this?
```
.format_na('-', subset=['col1','col2'])
.format('{:.2%}', na_rep='-', subset=['col3','col4'])
```
And the docstring for `format_na` rephrase to:
```
Format the text display value using default formatter but represent nan as `na_rep`.
For more advanced formatting, use Styler.format() with your custom formatter.
```

Reply:
I may not have been clear about my concern. It's fine that `na_format` overwrites the formatting for NA values. I'm concerned tht it overwrites the formatting for *non*-NA values. In my `.format("hi-{}".format).format_na('NA')` example, the NA values should be formatted as 'NA' and the non-NA values should be formatted as `hi-<value>`. But I suspect that right now the non-NA formatting is lost (though perhaps it's not).

Reply:
Not sure about adding an `na_rep` to the `.format` function... That's probably fine. I think it'd still be useful for users to have a way to control the default NA formatting at the table level.

But if we add an `na_rep` to `format`, then we wouldn't need a new `format_na` method, right?

Reply:
Sounds good, so a setting at the table level: `Styler.na_rep`
and the `def format(self, formatter, subset=None):` becomes 
`def format(self, formatter=None, na_rep=None, subset=None):`
drop `.format_na('-')`, use `.format(na_rep='-')` instead, right?

Reply:
I think that sounds correct. I'm not sure what the default should be, but probably just None (no special formatting for NA values).